From: Christian Kastner <ckk@debian.org>
Date: Tue, 7 Jul 2026 14:25:57 +0200
Subject: Use simple printf in usage

I noticed on the reproducible builds effort that the usage message
sometimes not not include examples. Looking at the affected executables,
this seems to be tied to the use of LOG(), which uses another thread
in the background for output.

This isn't really needed. printf does the same job and avoids threading
issues.

Not forwarding yet, as this is still just a hunch.
---
 examples/batched/batched.cpp          | 6 +++---
 examples/passkey/passkey.cpp          | 6 +++---
 tools/batched-bench/batched-bench.cpp | 6 +++---
 tools/tts/tts.cpp                     | 6 +++---
 4 files changed, 12 insertions(+), 12 deletions(-)

diff --git a/examples/batched/batched.cpp b/examples/batched/batched.cpp
index 830e45f..40c77e6 100644
--- a/examples/batched/batched.cpp
+++ b/examples/batched/batched.cpp
@@ -11,9 +11,9 @@
 #include <vector>
 
 static void print_usage(int, char ** argv) {
-    LOG("\nexample usage:\n");
-    LOG("\n    %s -m model.gguf -p \"Hello my name is\" -n 32 -np 4\n", argv[0]);
-    LOG("\n");
+    printf("\nexample usage:\n");
+    printf("\n    %s -m model.gguf -p \"Hello my name is\" -n 32 -np 4\n", argv[0]);
+    printf("\n");
 }
 
 int main(int argc, char ** argv) {
diff --git a/examples/passkey/passkey.cpp b/examples/passkey/passkey.cpp
index 8440a2b..a792bae 100644
--- a/examples/passkey/passkey.cpp
+++ b/examples/passkey/passkey.cpp
@@ -11,9 +11,9 @@
 #include <algorithm>
 
 static void print_usage(int, char ** argv) {
-    LOG("\nexample usage:\n");
-    LOG("\n    %s -m model.gguf --junk 250 --pos 90 --keep 32 --grp-attn-n 2 [--seed 1234]\n", argv[0]);
-    LOG("\n");
+    printf("\nexample usage:\n");
+    printf("\n    %s -m model.gguf --junk 250 --pos 90 --keep 32 --grp-attn-n 2 [--seed 1234]\n", argv[0]);
+    printf("\n");
 }
 
 int main(int argc, char ** argv) {
diff --git a/tools/batched-bench/batched-bench.cpp b/tools/batched-bench/batched-bench.cpp
index e2dcd0b..71f4d15 100644
--- a/tools/batched-bench/batched-bench.cpp
+++ b/tools/batched-bench/batched-bench.cpp
@@ -10,9 +10,9 @@
 #include <vector>
 
 static void print_usage(int, char ** argv) {
-    LOG("\nexample usage:\n");
-    LOG("\n    %s -m model.gguf -c 2048 -b 2048 -ub 512 -npp 128,256,512 -ntg 128,256 -npl 1,2,4,8,16,32 [-pps]\n", argv[0]);
-    LOG("\n");
+    printf("\nexample usage:\n");
+    printf("\n    %s -m model.gguf -c 2048 -b 2048 -ub 512 -npp 128,256,512 -ntg 128,256 -npl 1,2,4,8,16,32 [-pps]\n", argv[0]);
+    printf("\n");
 }
 
 // satisfies -Wmissing-declarations
diff --git a/tools/tts/tts.cpp b/tools/tts/tts.cpp
index 2a1bdcc..6278e36 100644
--- a/tools/tts/tts.cpp
+++ b/tools/tts/tts.cpp
@@ -69,9 +69,9 @@ const std::vector<std::string> k_colors = {
 };
 
 static void print_usage(int, char ** argv) {
-    LOG("\nexample usage:\n");
-    LOG("\n    %s -m model.gguf -p \"Hello!\"\n", argv[0]);
-    LOG("\n");
+    printf("\nexample usage:\n");
+    printf("\n    %s -m model.gguf -p \"Hello!\"\n", argv[0]);
+    printf("\n");
 }
 
 struct wav_header {
